C24_WMDE_Desktop_EN_02b (round2)

moiikana opened the pull request:
round 2 of the EN02 banner

- [x] I already donated 
- [x] soft close
- [x] progress bar
---
- [x] tests
- [X] alreadydonatedmodal code is still used by some precampaign banners and thus can't be removed until January 2025
- [x] contains fixes for payment icons and UoF graphics
---
- round 1 of this banner is here: https://github.com/wmde/fundraising-banners/pull/572


https://phabricator.wikimedia.org/T375601

moiikana committed:
Prepare campaign for C24_WMDE_Desktop_EN_02b (round2)

- based on C24_WMDE_Desktop_EN_02 (round1)
- Add fix for unstyled payment icons

https://phabricator.wikimedia.org/T375601

moiikana committed:
DesktopEN02b: Add AlreadyDonated feature

- won't have any modal, will just set a cookie (4 weeks no banner)  and close the banner
- rename emitted event to be more generic
- Adapt banner tests
- Add item to postcampaigncleanup plan

https://phabricator.wikimedia.org/T375601

moiikana committed:
DesktopEN02b: Add Progress Bar

- add component
- enable style for the progress bar

https://phabricator.wikimedia.org/T375601

moiikana committed:
DesktopEN02b: Add SoftClose feature

- add tests for it
- enable the style swatch

https://phabricator.wikimedia.org/T375601

moiikana commented:
still needs to be rebased properly

moiikana committed:
DesktopEN02b: Add UoF changes for infographics (02b)

- changes from this PR https://github.com/wmde/fundraising-banners/pull/563
- include fallback banner test in the current test run for 02b

moiikana commented:
contains the latest UoF changes from https://github.com/wmde/fundraising-banners/pull/563 :heavy_check_mark: 
- [x] banner ctrl
- [x] banner var
- [x] fallback banner

Sperling-0 commented:
⚠️ Rebase this PR from latest main.

To get the latest UoF working:

- For Desktop banners: `BannerCtrl.vue`, `BannerVar.vue`, `FallbackBanner.vue` needs to be updated. (If it is DE desktop include DE infographic and vice versa)
- For Mobile/ Ipad banners: `BannerCtrl.vue`, `BannerVar.vue` needs to be updated. (If it is DE mobile include DE infographic and vice versa)

moiikana commented:
rebased main,
contains the latest BIC changes https://phabricator.wikimedia.org/T377589

moiikana commented:
rebased main

Abban reviewed (approved):
Looks good, I just spotted a few things with the progress bar.

You should update the variable for the progress bar in the dark colour swatch to:

```--progress-bar-wrapper-background: #{$grey700};```

In the theme's ProgressBarAlternative.scss on line 69 you should increase the line height to 26px. It'll fix the alignment in the left text:

![Screenshot 2024-10-24 at 14 51 29](https://github.com/user-attachments/assets/f98f9e27-29b5-41b7-8f7b-3bbdc1baaad3)

moiikana committed:
DesktopEN02b: Fix progress bar background and spacing

- background should be dark in the dark mode
- add paragraph for second banner page
- make softclose ready for darkmode

Abban commented on banners/english/C24_WMDE_Desktop_EN_02b/styles/styles.scss:
```diff
@@ -0,0 +1,59 @@
+	$height: 34px
+);
+@use 'src/themes/Treedip/DonationForm/BubbleForm/SelectGroupBubbleImageLabel';
+@use 'src/themes/Treedip/DonationForm/BubbleForm/SelectGroupBubbleImageLabel_dark';
```
You can delete this import, the swatching improvements handle the payment icon colours differently now. If you want you can also delete the file and the imports from the older EN desktop banners

moiikana committed:
DesktopEN02b: Delete old _dark css approach

- this was used before we introduced swatches for dark mode

moiikana commented:
rebased because https://github.com/wmde/fundraising-banners/pull/597 was merged

Abban committed:
DesktopEN02b: Increase lightness of Treedip dark mode colours

The dark red and blue was failing WCAG contrast guidelines
so this brightens them just enough to pass AA.

moiikana committed:
DesktopEN02b: Move tests into own folder

this is for https://phabricator.wikimedia.org/T369276

moiikana commented:
rebased main

moiikana commented:
another main rebase

gbirke reviewed (approved):
I like the single-commit-for-single-feature-activation of this PR. It'll help later when looking at the history of a banner. To avoid making the titles look like we activated a feature for all campaigns we might want to prefix the title with the campaign name next time. If you like you could rebase-and-edit the existing commit titles, but we can also leave them as-is
